ldcoolp/curation/retrieve.py

```diff
@@ -34,32 +35,19 @@ def private_file_retrieve(url, filename=None, token=None, url_open=False,
     if isinstance(log, type(None)):
         log = log_stdout()
 
-    if not url_open:
-        if token:
-            opener = build_opener()
-            opener.addheaders = [('Authorization', f'token {token}')]
-            install_opener(opener)
-        else:
-            opener = build_opener()
-            install_opener(opener)
-
-        try:
-            urlretrieve(url, filename)
-        except HTTPError as error:
-            log.warning(f"Caught an HTTPError: {error}")
-            raise
-    else:
-        req = Request(url)
-        if token:
-            req.add_header('Authorization', f'token {token}')
+    headers = dict()
+    if token:
+        headers['Authorization'] = f'token {token}'
 
-        response = urlopen(req)
-        content = response.read()
-        print(url)
+    try:
+        h = requests.head(url, headers=headers)
+        h.raise_for_status()
 
-        f = open(filename, 'wb')
-        f.write(content)
-        f.close()
+        with requests.get(url, stream=True, headers=headers) as r:
+            with open(filename, 'wb') as f:
+                shutil.copyfileobj(r.raw, f)
+    except HTTPError as error:
```

Comment:
Should there be error handling code for errors related to low disk space? It's a very plausible scenario

Reply:
Yes, but that's out of scope for this PR as that's an entirely different issue in itself.